fda-model/alternatives/Alternative.cs

```diff
@@ -41,12 +50,12 @@ public class Alternative : Validation, IReportMessage, IProgressReport
             {
                 AlternativeResults nullAlternativeResults = new AlternativeResults(alternativeResultsID, analysisYears, periodOfAnalysis, false);
                 MessageEventArgs messageArguments = new MessageEventArgs(new Message("The discounting parameters are not valid, discounting routine aborted. An arbitrary results object is being returned"));
-                nullAlternativeResults.ReportMessage(nullAlternativeResults, messageArguments);
+                ReportMessage(this, messageArguments);
                 return nullAlternativeResults;
             }
             AlternativeResults alternativeResults = new AlternativeResults(alternativeResultsID, analysisYears, periodOfAnalysis);
             MessageEventArgs messargs = new MessageEventArgs(new Message("Initiating discounting routine."));
-            alternativeResults.ReportMessage(alternativeResults, messargs);
+            alternativeResults.ReportMessage(this, messargs);
```

Comment:
Should this line just ReportMessage itself or does it need to report it on the alternativeResults object? Probably doesn't really make a difference but everywhere else I see you just calling ReportMessage.

Reply:
does not need to report on alternative results. I think that many of the messages were not coming through because they were being reported on objects that were not registered. since we changed the class to no longer be static, we can just report on *this*